### ISSUE TITLE
Looking for a Maintainer !

### ISSUE THREAD

**Issue by asandikci** [MaintainTeam/Hypatia#1]
Unfortunately DivestOS project reached end of support. (https://divestos.org/pages/news) I think Hypatia is an important project beside of DivestOS. It would be good to keep it maintained.

I don't have enough knowledge/time to maintain this project but if someone wants to maintain this project please specify these:
- Why do you want to maintain this project?
- How and with doing what you can keep this project maintained 
- What is your knowledge about cyber-sec & android development (any project/link that prove that knowledge)

You can type in here or in `Maintenance Requests` Channel in [Matrix Room](https://matrix.to/#/!TZsSAfjDhuQnpfEbIc:matrix.org?via=matrix.org)

**Comment by licaon-kter** [comment]
The first step would be to provide updated databases, as the app still works, but we need freshly generated data, hosted somewhere so users can point the URL to it.

Is the procedure clear at least?

**Comment by asandikci** [comment]
Related: https://github.com/MaintainTeam/HypatiaDatabases/issues/1

**Comment by steadfasterX** [comment]
stumbled over this one just now... 

I started a fork of it, too: https://codeberg.org/AXP-OS/packages_apps_LoveLaceAV
I already started the signature db handling which turned out to be not as easy as thought first. mainly bc several parts of the "how" isn't clear or documented. but I am pretty far already. I also was in contact with Tad (the creator of Hypatia) and got all required scripts, latest released database and information so I am pretty sure to get things done. Just a a matter of time.

atm I have other priorities but if there is a maintainer willing to do the work I can share what I have and/or "just" providing the signature database updates. It is still a WIP but I got far already: https://github.com/sfX-android/automation_scripts/tree/ansible/roles/hypatiadb

as you can see I automate that with Ansible and updates and the results will be available over `dbsignatures.axpos.org` (once ready).

[dbsig.html.txt](https://github.com/user-attachments/files/18780456/dbsig.html.txt)

also see: https://codeberg.org/AXP-OS/tools_lovelaceav_convertdb

**Comment by asandikci** [comment]
@steadfasterX Thanks for the sources and information. Tad contacted with me also. Sent scripts & your ansible script links too. Since I'm not familiar with Ansible I've used just CI (so pure bash) to generate databases for now. But couldn't find a spare time to handle all signature sources. (Current CI in `HypatiaDatabasesTests`* repo includes only [these](https://github.com/MaintainTeam/HypatiaDatabasesTests/actions/runs/13350827967/job/37287092351) ). Yours seems more complete. You've included all of them if I didn't get wrong. Good job. 

*[blocker](https://github.com/MaintainTeam/HypatiaDatabases/issues/4#issuecomment-2645904920) to merge into actual `HypatiaDatabases` repo

**Comment by TheAlmightyOgreLord** [comment]
It looks like we almost have a team here. Is there a reason why the database updates couldn't be handled by a free tier oracle VPS? If hosting is the issue which it seems so right now... I'm still getting my workstation up and running but my eyes are glued to this project. With how difficult it is to keep devices secure these days (due to constant rebasing to new android releases and discontinuation of OS security patches) this is a very important security layer I think.

**Comment by steadfasterX** [comment]
> [@steadfasterX](https://github.com/steadfasterX) Thanks for the sources and information. Tad contacted with me also. Sent scripts & your ansible script links too. Since I'm not familiar with Ansible I've used just CI (so pure bash) to generate databases for now. But couldn't find a spare time to handle all signature sources. (Current CI in `HypatiaDatabasesTests`* repo includes only [these](https://github.com/MaintainTeam/HypatiaDatabasesTests/actions/runs/13350827967/job/37287092351) ). Yours seems more complete. You've included all of them if I didn't get wrong. Good job.
> 
> *[blocker](https://github.com/MaintainTeam/HypatiaDatabases/issues/4#issuecomment-2645904920) to merge into actual `HypatiaDatabases` repo

ok so for me using the github CI/CD is a no-go (when it comes to **signing**) since the acquisition by M$. yea call me paranoid but anyways. I haven't added all sources yet but the majority, yes. there is still some time needed to finalize the rest though.

as long as the db signatures are signed properly they can be hosted wherever we like imho. I was just offering to do it but whatever works.

any motivation to rename the project or do you want to keep it as it is? If you do not like my naming (LoveLaceAV) I would still vote for at least adding "AV" to get a better feeling what this is about.

**Comment by TheAlmightyOgreLord** [comment]
I vote for something like Imperator to complement the meaning of Hypatia but lovelace is also clever

**Comment by asandikci** [comment]
> ok so for me using the github CI/CD is a no-go (when it comes to **signing**) since the acquisition by M$. yea call me paranoid but anyways

I agree. I'm paranoid too. I don't plan to use same signing key in different providers (see https://github.com/MaintainTeam/HypatiaDatabases/issues/5)


> as long as the db signatures are signed properly they can be hosted wherever we like imho. I was just offering to do it but whatever works.

👍🏻 


> any motivation to rename the project or do you want to keep it as it is?

I didn't think about it yet. I find AV suffix unnecessary. But re-branding should be discussed after someone found for maintain* this app

*I mean developing app itself, solving bugs, new features etc. in Hypatia Android App. Database side doesn't have to be actively maintained after proper CI/scripts done by anyone once

**Comment by IzzySoft** [comment]
@asandikci would you consider making at least clear with the app description in Fastlane (`short_descrition.txt`: append "(fork)"; `full_description.txt`: append something like "This is a fork of the original (and discontinued) app.", maybe even "This is a fork of the original (and discontinued) app, which is still looking for maintainers.")? Texts can be updated again should a maintainer have picked up. Then, the graphics should be updated as well to make the difference clear, maybe even the name should be changed then.

**Comment by TheAlmightyOgreLord** [comment]
Dang it, you guys are alluring me to get up and running. I have some really good ideas for this project. Hang in there

**Comment by DrDustwolf** [comment]
Thanks a lot for Forking Hypatia, it certainly would be a great loss if the programm would have been abandoned for good!

Do guys already have a new working URL and Signing Key for the Database?

BTW: I like the new name Imperator or ImperatorAV more than LovelaceAV.

**Comment by asandikci** [comment]
> [@asandikci](https://github.com/asandikci) would you consider making at least clear with the app description in Fastlane (`short_descrition.txt`: append "(fork)"; `full_description.txt`: append something like "This is a fork of the original (and discontinued) app.", maybe even "This is a fork of the original (and discontinued) app, which is still looking for maintainers.")? Texts can be updated again should a maintainer have picked up. Then, the graphics should be updated as well to make the difference clear, maybe even the name should be changed then.

@IzzySoft Added. Is it enough to push new files to default branch or should I release a new version too (to be updated on your repo)

**Comment by asandikci** [comment]
> Do guys already have a new working URL and Signing Key for the Database?

@DrDustwolf You have 3 option **for now**
- Current default Database in this fork  
  - `https://raw.githubusercontent.com/MaintainTeam/HypatiaDatabases/refs/heads/main/`
  - `5298C0C0C3E73288`
  - [HypatiaDatabases](https://github.com/MaintainTeam/HypatiaDatabases) Generated Locally **and not up-to-date** [2025 Jan]
- **TEST** Database
  - `https://maintainteam.github.io/HypatiaDatabasesTests/`
  - `C335CDA7A1C0F66B`
  - [HypatiaDatabasesTests](https://github.com/MaintainTeam/HypatiaDatabasesTests) Generated with CI and up-to-date [each midnight]
  - More Information: https://maintainteam.github.io/HypatiaDatabasesTests/
  - <sub>I'll merge this repo to main (HypatiaDatabases) repo after both https://github.com/MaintainTeam/HypatiaDatabases/issues/5 and https://github.com/MaintainTeam/HypatiaDatabases/issues/4 has been resolved</sub>
- Other Databases generated from someone or you can always generate yourself :)

**Comment by IzzySoft** [comment]
> Added. Is it enough to push new files to default branch or should I release a new version too (to be updated on your repo)

Thanks! To answer your question:

```
$ iod repo fastlane org.maintainteam.hypatia
org.maintainteam.hypatia: check if repo contains FUNDING.yml
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/.github'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/.github/contents/'
org.maintainteam.hypatia: no FUNDING.yml detected.
org.maintainteam.hypatia: calling 'getFastlaneMeta(github,[host:github.com,owner:MaintainTeam,repo:Hypatia,path:/fastlane/metadata/android])'
org.maintainteam.hypatia: FastlaneFeatures title,shortdesc,fulldescMD,changelogs,icon,featureGraphic,screenshots
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fcs'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fde'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fde%2Fimages'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fde%2Fimages%2FphoneScreenshots'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US%2Fchangelogs'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US%2Fimages'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US%2Fimages%2FphoneScreenshots'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fes'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fes%2Fimages'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Ffr'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Ffr%2Fimages'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Ffr%2Fimages%2FphoneScreenshots'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fpl-PL'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fpt-BR'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fpt-BR%2Fimages'
org.maintainteam.hypatia: looking for 'https://api.github.com/repos/MaintainTeam/Hypatia/contents/fastlane%2Fmetadata%2Fandroid%2Fru'
org.maintainteam.hypatia: checking locale 'cs'
org.maintainteam.hypatia: checking locale 'de'
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/de/phoneScreenshots/01-Main.png' as it's not newer than ours.
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/de/phoneScreenshots/02-Detection.png' as it's not newer than ours.
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/de/phoneScreenshots/03-Notification.png' as it's not newer than ours.
org.maintainteam.hypatia: checking locale 'en-US'
org.maintainteam.hypatia: replacing 'metadata/org.maintainteam.hypatia/en-US/full_description.txt' with fromMD
org.maintainteam.hypatia: updating 'metadata/org.maintainteam.hypatia/en-US/short_description.txt'
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/en-US/phoneScreenshots/01-Main.png' as it's not newer than ours.
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/en-US/phoneScreenshots/02-Detection.png' as it's not newer than ours.
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/en-US/phoneScreenshots/03-Notification.png' as it's not newer than ours.
org.maintainteam.hypatia: checking locale 'es'
org.maintainteam.hypatia: checking locale 'fr'
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/fr/phoneScreenshots/01-Main.png' as it's not newer than ours.
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/fr/phoneScreenshots/02-Detection.png' as it's not newer than ours.
org.maintainteam.hypatia: skipping 'repo/org.maintainteam.hypatia/fr/phoneScreenshots/03-Notification.png' as it's not newer than ours.
org.maintainteam.hypatia: checking locale 'pl-PL'
org.maintainteam.hypatia: checking locale 'pt-BR'
org.maintainteam.hypatia: checking locale 'ru'
````

😜

> You have 3 option for now

All of them MS bound I see, so we'll need the `NonFreeNet` anti-feature I guess? Well, not **strongly** needed due to 3, if that's configurable inside the app. But honestly, that would be no real option for many. Maybe consider a mirror at Codeberg.org (they also have "Codeberg Pages" for static content), and maybe one day even make that the primary?

**Comment by Inhishonor** [comment]
I would like to help maintain this app. Unfortunately, I am still learning android development, so I won't be much help code wise, at least for a little bit, but I do have experience with github builds, and I can triage and test issues. I also, could help with documentation as well, if needed. Thanks for your consideration!

**Comment by DwainZwerg** [comment]
@asandikci When I want to try the TEST Database, my device reports that the databases are not available, even though I downloaded them beforehand. EDIT: https://codeberg.org/MaintainTeam/HypatiaDatabases/ (meaning works without ‘test’)

![Image](https://github.com/user-attachments/assets/40deefb4-a0b5-4d03-9a1a-74847e1c5c8e)

**Comment by asandikci** [comment]
@DwainZwerg extended database not generated. Thats why you get error. Track https://github.com/MaintainTeam/HypatiaDatabases/issues/6

**Comment by DwainZwerg** [comment]
No. That's not the problem: https://raw.githubusercontent.com/MaintainTeam/HypatiaDatabases/refs/heads/main/ does not include the extended database. But it works there (meaning there is only a message for the extended database, but the scan runs through there). With https://maintainteam.github.io/HypatiaDatabasesTests/ the scan does not run at all, as if the other databases were completely empty. With https://maintainteam.github.io/HypatiaDatabases/ it works again.

**Comment by TheAlmightyOgreLord** [comment]
what I think @asandikci means is extended database files are not generated yet on the fileserver(s)

**Comment by asandikci** [comment]
@DwainZwerg oh I got you now. It's probably because of you didn't changed signing key for Tests repo. Or anything related to cache. Since Tests database uses a **different** temporary [signing key](https://maintainteam.github.io/HypatiaDatabasesTests/) (that leaked to GitHub) I've been set up a new self-hosted CI to sign databases with our main sign key. Then merged to HypatiaDatabases repos from Tests repo. You can use new databases without any problems. Tests repo will be delete soon, and new version of app will warn you if using an old database (like tests repo).

For more organized info please see [Signature Databases](https://github.com/MaintainTeam/Hypatia?tab=readme-ov-file#signature-databases) section in Readme

and please open a new issue for any other problem, marking as off-topic above comments

**Comment by asandikci** [comment]
> All of them MS bound I see, so we'll need the `NonFreeNet` anti-feature I guess? Well, not **strongly** needed due to 3, if that's configurable inside the app. But honestly, that would be no real option for many. Maybe consider a mirror at Codeberg.org (they also have "Codeberg Pages" for static content), and maybe one day even make that the primary?

@IzzySoft I've actually set up a mirror and made Codeberg as primary (in dev branch) but when I'm testing faced with some errors. Track in: #11

**Comment by IzzySoft** [comment]
@asandikci just let me know when I shall "switch the URLs". I'm with Codeberg since (before) they've launched, I also use pages there (though rarely). Apart from small hick-ups every now and then (usually caused by some DDOS), I've never experienced any issues. The few unrelated ones where usually quickly addressed by Codeberg staff, which is friendly and communicative.

**Comment by asandikci** [comment]
> [@asandikci](https://github.com/asandikci) just let me know when I shall "switch the URLs". I'm with Codeberg since (before) they've launched, I also use pages there (though rarely). Apart from small hick-ups every now and then (usually caused by some DDOS), I've never experienced any issues. The few unrelated ones where usually quickly addressed by Codeberg staff, which is friendly and communicative.

It was a temporary thing I guess. Everything works now. Preparing a new release with default codeberg url soon 🥳 

I'll be pleasure if anyone can test this PR #38 

(pinging some potential testers, thanks in advance. @steadfasterX @Inhishonor @TheAlmightyOgreLord )

**Comment by IzzySoft** [comment]
> It was a temporary thing I guess.

Recently they get frequently attacked it seems (DDOS and such). As if someone doesn't like their success (wtf, a community run forge without capitalists inside? Must not be! They don't give us data!11!) I donate to them regularly. When those big attacks started, I decided that's not enough and became a full member (choosing the "double my membership fee" option). They shall be here to stay.

> Preparing a new release with default codeberg url soon 🥳

🥳 

Sorry that I cannot test (not using the app myself, and am already drowning in tasks)…

**Comment by Captn138** [comment]
Hi all !
If the LoveLaceAV becomes the spiritual successor of Hypatia, could it be possible to notify users of Hypatia when a first release of LoveLaceAV is available with a notification for example ?

**Comment by asandikci** [comment]
> The first step would be to provide updated databases, as the app still works, but we need freshly generated data, hosted somewhere so users can point the URL to it. @licaon-kter 

https://github.com/MaintainTeam/Hypatia?tab=readme-ov-file#signature-databases 🙌🏻
<br>

> any motivation to rename the project or do you want to keep it as it is? If you do not like my naming (LoveLaceAV) I would still vote for at least adding "AV" to get a better feeling what this is about. @steadfasterX 

> I vote for something like Imperator to complement the meaning of Hypatia but lovelace is also clever @TheAlmightyOgreLord 

--> Could we please move to: https://github.com/MaintainTeam/Hypatia/discussions/14 🤗 
<br>

>  If the LoveLaceAV becomes the spiritual successor of Hypatia, could it be possible to notify users of Hypatia when a first release of LoveLaceAV is available with a notification for example ? @Captn138

LoveLaceAV is maintained by @steadfasterX

**Comment by steadfasterX** [comment]
> LoveLaceAV is maintained by [@steadfasterX](https://github.com/steadfasterX)

see my new comment in #14

**Comment by Siradankullanici** [comment]
I can boost Hypatia project with my signature and heuristics and can become one of that maintainers. I working on HydraDragonAntivirus project which is pretty complex right now. There open source tools for making antivirus in past for Android but then they become closed source like Koodus. We need open source antivirus for android.

**Comment by shivagnihotri** [comment]
So where to find a friendly and supportive maintainer to keep this important app alive? As for me, I'd love someone like @d4rken to be on board.

**Comment by asandikci** [comment]
> I can boost Hypatia project with my signature and heuristics and can become one of that maintainers. I working on HydraDragonAntivirus project which is pretty complex right now.

@Siradankullanici  I'm glad to hear that! Could you please contact in Matrix with us? 

> We need open source antivirus for android.

I totally agree 🙌🏻 That's why We're searching for a maintainer


> So where to find a friendly and supportive maintainer to keep this important app alive? As for me, I'd love someone like @d4rken to be on board.

I didn't contacted with Matthias but It would be good for the future of Hypatia if he can maintain the app

**Comment by d4rken** [comment]
I don't have time for an unpaid extra project at the moment. Sorry.

**Comment by shivagnihotri** [comment]
> I can boost Hypatia project with my signature and heuristics and can become one of that maintainers. I working on HydraDragonAntivirus project which is pretty complex right now. There open source tools for making antivirus in past for Android but then they become closed source like Koodus. We need open source antivirus for android.

Yes, please do and submit a pull request - when people see that the code in this repository is being maintained, they are more likely to join.

I undestand that @d4rken is very busy, check out his most awesome app [SD MAID SE FOSS](https://github.com/d4rken-org/sdmaid-se)! :)

**Comment by shivagnihotri** [comment]
Any news on finding someone to revive Hypatia?

**Comment by LUGKILL2** [comment]
I'd like to be one of the teammates for Hypatia, I can contribute some effort to help with some tasks. I don't have specific knowledge yet to be a maintainer, but if the devs would like to have me, I'd be open. In the process, I will be learning the stuff that is required to be a good effective team member.

**Comment by asandikci** [comment]
> Any news on finding someone to revive Hypatia?

Not yet unfortunately. Anyone interested with maintaining (or contributing to) Hypatia is invited to our Matrix chat 🤗 
https://matrix.to/#/!TZsSAfjDhuQnpfEbIc:matrix.org?via=matrix.org

**Comment by mjryan253** [comment]
Hi, I'm a 20 year IT veteran who has run several large scale development efforts. I love Hypatia, I just found it and I believe I could make some major ui/ux and backend steps forward. I'm going to branch from the main branch and see what I can update. If I find something improved, I'll be happy to submit a pr for review. 👍

**Comment by Inhishonor** [comment]
Hello everyone,
I will be maintaining this app. It will be all done through this repo, and I am hoping to get this app going again. A big thank you for @asandikci for picking up this app, and keeping it going. If anyone still wants to help, in any capacity (documentation, translation, testing, or development), it would be very much appreciated. I am also very willing to help anyone who is having trouble.

Thank you!

**Comment by Inhishonor** [comment]
> Hi, I'm a 20 year IT veteran who has run several large scale development efforts. I love Hypatia, I just found it and I believe I could make some major ui/ux and backend steps forward. I'm going to branch from the main branch and see what I can update. If I find something improved, I'll be happy to submit a pr for review. 👍

Hello, @mjryan253 That would be great! I would recommend doing your work off the dev branch, as I recently rewrote everything in Kotlin from Java, and those changes haven't been merged into stable yet. Let me know if I can help.
